vscode-extensions/vscode-spring-boot/lib/explorer/structure-tree-manager.ts

```diff
@@ -17,14 +16,20 @@ export class StructureManager {
             if (node && node.getReferenceValue) {
                 const reference = node.getReferenceValue();
                 if (reference) {
-                    // Reference is a specific URL that should be passed to java.open.file command
-                    commands.executeCommand('java.open.file', reference);
+                    const uri = Uri.parse(reference);
+                    if (uri.scheme === 'jdt') {
```

Comment:
I think we should not distinguish between those different URLs here and assume instead that we just use `vscode.open` to open the URI - and that's it. The language server can then produce the newly introduced `spring-boot-ls` URLs for all the references to JSON catalog files and we are done.